packages/cli/src/deploy/ensureWorldFactory.ts

```diff
@@ -17,7 +17,7 @@ export const accessManagementSystemDeployedBytecodeSize = size(
 );
 export const accessManagementSystemBytecode = encodeDeployData({
   bytecode: accessManagementSystemBuild.bytecode.object as Hex,
-  abi: [],
+  abi: [] as Abi,
```

Comment:
why do we have to cast this? I would expect TS to be able to infer this in the call to `encodeDeployData`

Reply:
`abi: []` is now not allowed for `encodeDeployData`. I think `[]` is not the abi expected here. I'll look into this.

```
src/deploy/ensureWorldFactory.ts:18:64 - error TS2345: Argument of type '{ bytecode: `0x${string}`; abi: never[]; }' is not assignable to parameter of type 'never'.

18 export const accessManagementSystemBytecode = encodeDeployData({
                                                                  ~
19   bytecode: accessManagementSystemBuild.bytecode.object as Hex,
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
20   abi: [],
   ~~~~~~~~~~
21 });
   ~
```

Reply:
The encodeDeployData function does the task only if the contract has constructor args. It appears to have started rejecting abis with no effects at the type level? The logic of the function hasn't changed, so its behavior remains the same as before.

Alternatively, we can bypass this function:

```diff
diff --git a/packages/cli/src/deploy/ensureWorldFactory.ts b/packages/cli/src/deploy/ensureWorldFactory.ts
index 1f2913a8..f45cbed8 100644
--- a/packages/cli/src/deploy/ensureWorldFactory.ts
+++ b/packages/cli/src/deploy/ensureWorldFactory.ts
@@ -15,10 +15,7 @@ import { Contract } from "./ensureContract";
 export const accessManagementSystemDeployedBytecodeSize = size(
   accessManagementSystemBuild.deployedBytecode.object as Hex
 );
-export const accessManagementSystemBytecode = encodeDeployData({
-  bytecode: accessManagementSystemBuild.bytecode.object as Hex,
-  abi: [] as Abi,
-});
+export const accessManagementSystemBytecode = accessManagementSystemBuild.bytecode.object as Hex;
 export const accessManagementSystem = getCreate2Address({
   from: deployer,
   bytecode: accessManagementSystemBytecode,
```

Reply:
seems fine to leave the empty ABI approach!